R/glimpse.R

```diff
@@ -169,14 +172,20 @@ format_glimpse.list <- function(x, ..., .inner = FALSE) {
 
 #' @export
 format_glimpse.character <- function(x, ...) {
-  encodeString(x, quote = '"')
+  out <- encodeString(as.character(x), quote = '"')
+  out[is.na(x)] <- crayon_red(NA)
+
+  out
 }
 
 #' @export
 format_glimpse.factor <- function(x, ...) {
   if (any(grepl(",", levels(x), fixed = TRUE))) {
-    encodeString(as.character(x), quote = '"')
+    out <- encodeString(as.character(x), quote = '"')
   } else {
-    format(x, trim = TRUE, justify = "none")
+    out <- format(x, trim = TRUE, justify = "none")
   }
+  out[is.na(x)] <- crayon_red(NA)
```

Comment:
This looks like a pattern that we use in many places: assign a red `NA` string to a vector where another vector has `NA` values.

- Should this be a function?
- Should this be `crayon_red(NA)` or `crayon_red(out[is.na(x)])` instead?